components/playlist/renderer/playlist_render_frame_observer.cc

```diff
@@ -76,6 +79,10 @@ PlaylistRenderFrameObserver::GetMediaResponder() {
 }
 
 void PlaylistRenderFrameObserver::RunScriptsAtDocumentStart() {
+  if (!is_playlist_enabled_callback_.Run()) {
```

Comment:
It's not clear why `is_playlist_enabled_callback_` is a callback, not just `bool`.
In fact we can't efficiently apply different `playlist_enabled` value without frame reloading.

Reply:
Also this approach creates `PlaylistRenderFrameObserver` even the matching setting is disabled.
1) it takes some resources and CPU time on frame creation/destruction
2) in future changes this check could be missed in some new method that would result a similar bug.

IMHO, a reliable way is just to check it **before** creating `PlaylistRenderFrameObserver`.

Reply:
Even though the `PlaylistRenderFrameObserver` is not sensitive to changes to the `playlist_enabled` flag, there are cases where the frame observers are not being recreated on a page refresh, which means that for open tabs, changes in the `Enable Playlist` toggle (in `brave://settings`) are not reflected — hence we decided to stick with the current approach.